restore/action.yml

```diff
@@ -15,6 +15,10 @@ inputs:
     description: 'An optional boolean when enabled, allows windows runners to restore caches that were saved on other platforms'
     default: 'false'
     required: false
+  fail-on-cache-miss:
+    description: 'Fail the workflow if the cache is not found for the primary key'
```

Comment:
This would cause failure even if `restore-key` matched. I can imagine another case for failure should to happen when that also did not match. That would be true cache "miss".

Reply:
I agree. That's an implicit result if no cache is found for the primary key. I've updated the description to make it a bit more clear.

Reply:
Apologies I meant a different thing. What happen if a restore-key matched? The job should still proceed as there is a partial match. I can agree to fail if there is no match, but failing on partial match seems too strict to me. This looks like very small use case.
And we don't want to end up with many inputs like `fail-on-no-or-partial-match` and `fail-on-no-match` etc.